protocol/x/vault/keeper/shares.go

```diff
@@ -47,3 +56,62 @@ func (k Keeper) getTotalSharesIterator(ctx sdk.Context) storetypes.Iterator {
 
 	return storetypes.KVStorePrefixIterator(store, []byte{})
 }
+
+// MintShares mints shares of a vault for `owner` based on `quantumsToDeposit` by:
+// 1. Increasing total shares of the vault.
+// 2. Increasing owner shares of the vault for given `owner`.
+func (k Keeper) MintShares(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+	owner string,
+	quantumsToDeposit *big.Int,
+) error {
+	// Quantums to deposit should be positive.
+	if quantumsToDeposit.Cmp(big.NewInt(0)) <= 0 {
+		return types.ErrInvalidDepositAmount
+	}
+	totalShares, exists := k.GetTotalShares(ctx, vaultId)
+	existingTotalShares := totalShares.NumShares.BigInt()
+	sharesToMint := big.NewInt(0)
+	if exists && existingTotalShares.Cmp(big.NewInt(0)) == 1 {
+		// Get vault equity.
+		equity, err := k.GetVaultEquity(ctx, vaultId)
+		if err != nil {
+			return err
+		}
+		// Don't mint shares if equity is non-positive.
+		if equity.Cmp(big.NewInt(0)) <= 0 {
+			return types.ErrNonPositiveEquity
+		}
+		// Mint `deposit * existing shares / vault equity` number of shares.
```

Comment:
nit: 
```suggestion
		// Mint `deposit equity* existing shares / existing vault equity` number of shares.
```

Reply:
updated to `deposit (in quote quantums) * existing shares / vault equity (in quote quantums)`